packages/node/src/local/updater.ts

```diff
@@ -24,3 +30,141 @@ export interface FlagConfigUpdater {
    */
   update(onChange?: (cache: FlagConfigCache) => Promise<void>): Promise<void>;
 }
+
+export class FlagConfigUpdaterBase {
+  protected readonly logger: Logger;
+
+  public readonly cache: FlagConfigCache;
+
+  public readonly cohortStorage: CohortStorage;
+  public readonly cohortFetcher?: CohortFetcher;
+
+  constructor(
+    cache: FlagConfigCache,
+    cohortStorage: CohortStorage,
+    cohortFetcher?: CohortFetcher,
+    debug = false,
+  ) {
+    this.cache = cache;
+    this.cohortFetcher = cohortFetcher;
+    this.cohortStorage = cohortStorage;
+    this.logger = new ConsoleLogger(debug);
+  }
+
+  protected async _update(
+    flagConfigs: Record<string, FlagConfig>,
+    isInit: boolean,
+    onChange?: (cache: FlagConfigCache) => Promise<void>,
+  ): Promise<void> {
+    let changed = false;
+    if (onChange) {
+      const current = await this.cache.getAll();
+      if (!Object.is(current, flagConfigs)) {
+        changed = true;
+      }
+    }
+
+    // Get all cohort needs update.
+    const cohortIds = CohortUtils.extractCohortIds(flagConfigs);
+    if (cohortIds && cohortIds.size > 0 && !this.cohortFetcher) {
+      throw Error(
+        'cohort found in flag configs but no cohort download configured',
```

Comment:
Good that this is checked - but what if they want to use only a subset of flags in a deployment, and that subset doesn't require cohorts 🤔 (cc: @bgiori)